### ISSUE TITLE
Search Function scrolls vertically but not horizontally to display ALL of a match.

### ISSUE THREAD

**Issue by pgdownload** [notepad-plus-plus/notepad-plus-plus#8804]
### Description of the Issue
Search Function scrolls vertically but not horizontally to display ALL of a match.

### Steps to Reproduce the Issue
1. Have a long document
2. Ensure some lines are longer than screen / window 
3. Start searching for a common term 

### Expected Behavior
The match is highlighted (Green) and you can see the whole match on the screen

IMO the FIND function should default to fully left scrolled every time the FIND NEXT button is hit (if the NEXT match after that one is not already on screen). Only if the next match is off the visible window should the screen be shifted to place it in the viewing area.

### Actual Behavior
If just 1 pixel of the match is visible on the screen then the line is not scrolled horizontally to display it.
Horizontal scrolling is done if the match is completely off the screen.

![image](https://user-images.githubusercontent.com/26446313/92048825-120a5600-edcc-11ea-91a3-fbf958d98177.png)

**Comment by dail8859** [comment]
Please post your debug information as requested by the issue template.

**Comment by sasumner** [comment]
Closing issue due to refusal to provide _Debug Information_.

**Comment by dail8859** [comment]
> Not really a fan of devout dogma

The debug information is _incredibly_ useful to developers.

> But at least have a go before asking others to fill out unnecessary paper work.

I did try it. I could not reproduce it.

>  fill out unnecessary paper work.

It takes exactly 3 clicks to copy the debug information. This is not burdensome.

---

You'd be surprised how many times people have issues that you try to replicate only to find out they are on a version that is years old.

**Comment by sasumner** [comment]
@dail8859 

Please give me insight into why so many people delete the _Debug Information_ out of the template before submitting an issue.

I even tried to improve the instructions in the template in that area recently, apparently to no avail.

It may take more effort for a user to delete out that section in the template than to actually provide the info.

**Comment by dail8859** [comment]
@sasumner I wish I knew 🤷

**Comment by sasumner** [comment]
I'm not really pursuing this issue, due to the OP not wanting to be cooperative, but:

>Expected Behavior
The match is highlighted (Green) and you can see the whole match on the screen

_Find Next_ doesn't highlight the match in green, unless of course one has changed colors away from defaults (unknown in this case); the green shown above seems more like the _Smart Highlight_ green.

**Comment by pgdownload** [comment]
First up, apologies. I help out on a few forums myself and nothing more irritating than people getting irritated with others who are just trying to help out. Hopefully you can reopen the thread (as the issue does exist in a recent version.) Debug Info as requested below (Normally I'd provide it but the example seemed easily reproducible, apparently not though)

Notepad++ v7.8.9   (64-bit)
Build time : Jul 15 2020 - 20:31:49
Path : C:\Program Files\Notepad++\notepad++.exe
Admin mode : OFF
Local Conf mode : OFF
OS Name : Windows 10 Pro for Workstations (64-bit) 
OS Version : 1909
OS Build : 18363.900
Current ANSI codepage : 1252
Plugins : CodeAlignmentNpp.dll mimeTools.dll NppConverter.dll NppExport.dll

**Comment by pgdownload** [comment]
> _Find Next_ doesn't highlight the match in green, unless of course one has changed colours away from defaults (unknown in this case); the green shown above seems more like the _Smart Highlight_ green.

The green was a furphy. Regardless of the colour of the word FIND NEXT  behaves as I mentioned above. I've attached a short video to better explain. Here I'm searching on 'ER'. Notice the scroll bar at then bottom. It moves right (as expected) to show a subsequent search but then doesn't move back left enough to highlight the next find.

[Notepad.zip](https://github.com/notepad-plus-plus/notepad-plus-plus/files/5178696/Notepad.zip)

**Comment by sasumner** [comment]
Okay, I can reproduce, so I'll reopen.

**Comment by sasumner** [comment]
There are simpler cases I'm sure to reproduce, but here's one I found:

```
Dream other work press heard cat; stead way, anger event reason else thick read history, flow word man brown shape measure deal ice poem decide buy; six, shoe burn chief connect, last; syllable gather may many she, well among finish hundred period change turn total eat multiply prove boy develop believe whether reach multiply sheet, capital, village fair; numeral food soldier range, work sharp my, surface step locate wide girl young step imagine feed claim smell, near, story create paint least ship agree open touch prepare problem require join collect been, could bone log strong had suggest glad north possible steel do depend, summer might sea fell force five hunt; hurry especially river during found join danger low cell thought true brought; thus, pretty grand steel check tube find loud man paragraph animal more notice radio soon; feel strong hold such, system since like notice excite; shine war speak symbol ever draw, course gather problem; roll receive any also leave need, feet send be begin decide love, slip sing appear settle, fat people our, band read phrase; road smile yes rise south moon voice fight planet, flat, tell broke own process note experiment imagine here, real short, branch mother deep game collect what least stop law proper crowd page list are box equal, where, eat, skin, face double, first measure chord sense simple fair, sing paragraph, push opposite stood compare master what which similar know least happen again death position, quart; fight bat surprise shoe serve bottom, work flow said are; point north floor, heard late make new tool light colony at duck kill call under yes famous energy step collect nor solve together exercise feel past prove raise crowd cost clean, teeth division about don't your select gold spoke watch whether than card visit; mark; person.
Over, kill motion skill kill red natural material people, chart press natural drive bank produce triangle; provide never silent hour play, rule, death try strange locate segment school whose substance show; one best rain set gentle pound tree; cool think against product pass insect catch to quick crop with fast red locate salt voice total slave train make bit numeral gave children syllable why even mine caught; track product hear friend crowd, grew reach watch particular went week bat certain gold took dollar spell red noise track street; world country tail born here meat farm sea got for whole; group sit suffix order beat insect day paper dog book eight type suit divide duck edge, dad meant add cook mass wash period, modern eye stream would weather shore wide eat poem brown dollar begin bit smell speak, hour; egg at area dream mix way direct, plan determine; little push sat sister your ready figure lone wild shoe apple mass row close block note compare sell south locate north, station lost less row, chart lot black; found stead material, strange bat flower symbol hundred insect air neck happen, line noun stay difficult original create ocean train especially, success view neighbor trip, shoe major son glad saw blue kill want spend crease north station do yellow near distant pick feet island world, could teeth wall develop mile cold product less cry skill cool appear, send happen double fire break crop moment quick now if gray four, pattern name company north me left scale between guess, got just week stood wish lie thought do, wonder, girl drop voice, port baby may home hot neck start; winter is; man fact head lie, dollar want buy original, run, molecule put instant bat north, perhaps few were duck in plural stick radio young foot say game excite.
```

Word wrap off.  Search for `ther`.  Found a few times on line 1.  When you press _Find Next_ and the search goes from line 1 to line 2, the first match on line 2 exhibits the problem.

**Comment by pgdownload** [comment]
> Okay, I can reproduce, so I'll reopen.

Thanks. Good luck.

**Comment by sasumner** [comment]
Changes from this issue released in Notepad++ version 7.9
